src/gt4py/next/program_processors/runners/dace_fieldview/transformations/simplify.py

```diff
@@ -0,0 +1,979 @@
+
+    Args:
+        sdfg: The SDFG to process.
+        repl: Maps the name of the symbol to the value it should be replaced with.
+        validate: Perform validation at the end of the function.
+        validate_all: Perform validation also on intermediate steps.
+
+    Note:
+        This function expects that the symbol is replaced with a value. Thus it is
+        allowed to use this function to rename symbols. Furthermore, the function
+        will remove the substituted symbols from the SDFG, by calling
+        `SDFG.remove_symbol()`, which removes the symbol from `.symbols` `dict` as
+        well as from the symbol map in case of a nested SDFG.
+    """
+    # For consistency reasons we have to traverse them twice, however, we should
+    #  actually implement a custom replace path.
+    nested_sdfgs = list(sdfg.all_sdfgs_recursive())
+
+    # Now replace the symbolic value with its actual value.
+    for nsdfg in nested_sdfgs:
+        nsdfg.replace_dict(repl)
+
+    if validate_all:
+        sdfg.validate()
+
+    # Now we have to clean up. The call above, actually targets that case that we want
+    #  to replace `symbol1` with `symbol2` not necessarily with a value. So the symbols
+    #  are still registered, and the symbol tables now contains
+    #  `{symbol_we_replaced: value_we_used}`, which we will now remove.
+    for nsdfg in nested_sdfgs:
+        nsymbols = nsdfg.symbols
+        for sname in repl.keys():
+            if sname in nsymbols:
+                nsdfg.remove_symbol(sname)
+
+    if validate:
+        sdfg.validate()
+
+
+def gt_reduce_distributed_buffering(
+    sdfg: dace.SDFG,
+) -> Optional[dict[dace.SDFG, dict[dace.SDFGState, set[str]]]]:
+    """Removes distributed write back buffers."""
+    pipeline = dace_ppl.Pipeline([DistributedBufferRelocater()])
+    all_result = {}
+
+    for rsdfg in sdfg.all_sdfgs_recursive():
+        ret = pipeline.apply_pass(sdfg, {})
+        if ret is not None:
+            all_result[rsdfg] = ret
+
+    return all_result
+
+
+@dace_properties.make_properties
+class GT4PyRednundantArrayElimination(dace_transformation.SingleStateTransformation):
+    """Special version of the redundant array removal transformation.
+
+    DaCe is not able to remove some redundant arrays. This transformation is specially
+    designed to remove these transient arrays. It matches two array `read` that is
+    read and written into `write`. The transformation applies if:
+    - `read` is a transient non view array.
+    - `write` has input degree 1 and output degree zero (sink node might be lifted).
+    - If `write` is non-transient, then `read` shall not have a upstream access node
+        that refers to the same global data.
+    - `read` has input degree larger than zero and output degree 1.
+    - `read` does not appear in any other state that is reachable from this state.
+    - `read` and `write` have the same dimensionality.
```

Comment:
I do not see the point of this?
Could you explain this more, this redundant removal stuff is super complex, although it looks simple.

To your real question, DaCe offers the `cover()` method.
However, it might not be able possible to verify if a subset is smaller, for example consider `i:(i+1)` is this smaller than `j:(j+100)` without knowing the actual value of both `j` and `i` you can not tell this.

Reply:
Maybe I am wrong, but what we are interested in here is the size of the subset, not the range. You can copy the subset `i:(i+1)` in to the destination `j:(j+1)`, whatever `i` and `j`.

Reply:
Now we are interested in the range of the subset.
Assume you have an array of length 100, the subsets `0:50` and `50:100` are both valid.
However, the subset `75:125`, although it has the same size as the other two is not valid.
This is mostly important if the two arrays have different sizes.

Reply:
I think I got it, thank you for the explanation.